store/copr/coprocessor.go

```diff
@@ -909,7 +970,10 @@ func (worker *copIteratorWorker) handleCopResponse(bo *Backoffer, rpcCtx *tikv.R
 				return nil, errors.Trace(err)
 			}
 		}
-		return worker.buildCopTasksFromRemain(bo, lastRange, task)
+		if worker.req.Streaming {
+			task.ranges = worker.calculateRetry(task.ranges, lastRange, worker.req.Desc)
```

Comment:
`lastRange` may be nil. You can check whether `lastRange` is nil in `calculateRetry()` and `calculateRemain()`. If it is nil, returns `task.ranges`.